grai-server/app/connections/models.py

```diff
@@ -64,3 +68,39 @@ class Meta:
         indexes = [
             models.Index(fields=["workspace", "namespace", "name"]),
         ]
+
+
+class Run(TenantModel):
+    tenant_id = "workspace_id"
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    connection = TenantForeignKey(
+        "Connection", related_name="runs", on_delete=models.CASCADE
+    )
+    status = models.CharField(max_length=255)
+    metadata = models.JSONField(default=dict)
+    workspace = models.ForeignKey(
+        "workspaces.Workspace",
+        related_name="runs",
+        on_delete=models.CASCADE,
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    started_at = models.DateTimeField(
+        blank=True,
+        null=True,
+    )
+    finished_at = models.DateTimeField(
+        blank=True,
+        null=True,
+    )
+    user = models.ForeignKey(
+        "users.User",
+        related_name="runs",
+        on_delete=models.CASCADE,
```

Comment:
This one is sort of tricky, the Run is sort of a historical catalogue right? In that case even if the user (or connection) were deleted we might prefer to keep a record of the run which was performed. What do you think? We could potentially set a default to missing or something like that?

Reply:
You are quite correct, I had even made the field nullable.  I have updated this to `models.SET_NULL`.